src/NUnitFramework/framework/Constraints/RangeConstraint.cs

```diff
@@ -33,26 +33,35 @@ namespace NUnit.Framework.Constraints
     /// </summary>
     public class RangeConstraint : Constraint
     {
-        private readonly IComparable from;
-        private readonly IComparable to;
+        private readonly object from;
+        private readonly object to;
 
         private ComparisonAdapter comparer = ComparisonAdapter.Default;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="RangeConstraint"/> class.
         /// </summary>
-        /// <remarks>from must be less than or equal to true</remarks> 
-        /// <param name="from">Inclusive beginning of the range. Must be less than or equal to to.</param>
-        /// <param name="to">Inclusive end of the range. Must be greater than or equal to from.</param>
+        /// <param name="from">Must be less than or equal to the <paramref name="to"/> value.</param>
+        /// <param name="to">Must be greater than or equal to the <paramref name="from"/> value.</param>
         public RangeConstraint(IComparable from, IComparable to) : base( from, to )
         {
-            // Issue #21 - https://github.com/nunit/nunit-framework/issues/21
-            // from must be less than or equal to to
-            if (comparer.Compare(from, to) > 0)
-                throw new ArgumentException( "from must be less than to" );
+            this.from = from;
+            this.to = to;
+            CompareFromAndTo();
```

Comment:
What happens in that scenario is that you are blocked from ever creating a range with those objects because they only way you possibly could is to 1) not check that `from < to` for *any* types or 2) pass the custom comparer in up front.